mirrord/cli/src/internal_proxy.rs

```diff
@@ -53,25 +56,42 @@ pub(crate) async fn proxy(watch: drain::Watch) -> Result<(), InternalProxyError>
 
     tracing::info!(?config, "internal_proxy starting");
 
-    if let Some(log_destination) = config.internal_proxy.log_destination.as_ref() {
-        let output_file = OpenOptions::new()
-            .create(true)
-            .append(true)
-            .open(log_destination)
-            .map_err(|e| InternalProxyError::OpenLogFile(log_destination.clone(), e))?;
-
-        let tracing_registry = tracing_subscriber::fmt()
-            .with_writer(output_file)
-            .with_ansi(false);
-
-        if let Some(log_level) = config.internal_proxy.log_level.as_ref() {
-            tracing_registry
-                .with_env_filter(EnvFilter::builder().parse_lossy(log_level))
-                .init();
-        } else {
-            tracing_registry.init();
-        }
-    }
+    // Setting up default logging for intproxy.
+    let log_destination = config
+        .internal_proxy
+        .log_destination
+        .as_ref()
+        .map(PathBuf::from)
+        .unwrap_or_else(|| {
+            let random_name: String = rand::thread_rng()
+                .sample_iter(&Alphanumeric)
+                .take(7)
+                .map(char::from)
+                .collect();
+            let timestamp = Utc::now().timestamp();
+
+            PathBuf::from(format!("/tmp/mirrord-intproxy-{timestamp}-{random_name}"))
+        });
+
+    let output_file = OpenOptions::new()
+        .create(true)
+        .append(true)
+        .open(&log_destination)
+        .map_err(|fail| {
+            InternalProxyError::OpenLogFile(log_destination.to_string_lossy().to_string(), fail)
+        })?;
+
+    let log_level = config
+        .internal_proxy
+        .log_level
+        .clone()
+        .unwrap_or_else(|| "mirrord=trace".to_string());
```

Comment:
```suggestion
        .clone()
        .unwrap_or_else(|| "mirrord=trace".to_string());
```
```suggestion
        .as_deref()
        .unwrap_or("mirrord=trace");
```
We don't need owned, I think